File: app/src/main/java/abandonedstudio/app/cookingapp/Database/DishDao.java
package abandonedstudio.app.cookingapp.Database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface DishDao {

    @Insert
    void insert(Dish dish);

    @Update
    void update(Dish dish);

    @Delete
    void delete(Dish dish);

    @Query("SELECT * FROM dishes WHERE category_id LIKE :categoryId ORDER BY dish_name COLLATE NOCASE")
    LiveData<List<Dish>> getAllDishesFromCategory(int categoryId);

    @Query("SELECT * FROM dishes ORDER BY dish_name COLLATE NOCASE")
    LiveData<List<Dish>> getAllDishes();

}

File: app/src/main/java/abandonedstudio/app/cookingapp/core/data/local/room/dish/DishDao.kt
package abandonedstudio.app.cookingapp.core.data.local.room.dish

import androidx.room.*
import kotlinx.coroutines.flow.Flow

@Dao
interface DishDao {

    @Insert
    suspend fun insert(dish: Dish)

    @Update
    suspend fun update(dish: Dish)

    @Delete
    suspend fun delete(dish: Dish)

    @Query("SELECT * FROM dishes WHERE category_id LIKE :categoryId ORDER BY dish_name COLLATE NOCASE")
    fun getAllDishesFromCategory(categoryId: Int): Flow<List<Dish>>

    @Query("SELECT * FROM dishes ORDER BY dish_name COLLATE NOCASE")
    fun getAllDishes(): Flow<List<Dish>>
}